graph/topo/bench_test.go

```diff
@@ -107,3 +121,71 @@ func BenchmarkUndirectedCyclesInGnp_10_half(b *testing.B) {
 func BenchmarkUndirectedCyclesInGnp_100_half(b *testing.B) {
 	benchmarkUndirectedCyclesIn(b, gnpUndirected_100_half)
 }
+
+func benchmarkSort(b *testing.B, g graph.Directed) {
+	for i := 0; i < b.N; i++ {
+		_, _ = Sort(g)
```

Comment:
Please remove the dogsled, and replace it with a check if you are worried about optimisation.

Reply:
Done!

I've done it like the following, but I'm not sure if it's what you had in mind given part of the dogsled still exists, so please let me know if you have any more feedback.

```go
		if _, err := Sort(g); err != nil {
			b.FailNow()
		}
```

Reply:
Also I've applied a similar change to the benchmarks for `DirectedCyclesIn` and `UndirectedCyclesIn` just in case.

Reply:
Prefer
```
		_, err := Sort(g)
		if err != nil {
			b.FailNow()
		}
```

Reply:
> Prefer
> 
> ```
> 		_, err := Sort(g)
> 		if err != nil {
> 			b.FailNow()
> 		}
> ```

@kortschak Done! Force-pushed in [9135c94](https://github.com/gonum/gonum/pull/1948/commits/9135c941a4c0ee08ca9edbb194c0818c09e92db8).